scripts/deploy_on_ci.R

```diff
@@ -0,0 +1,122 @@
+# Set the account info for deployment.
+server <- "shinyapps.io"
+account <- Sys.getenv("SHINYAPPS_NAME", "unleash-shiny")
+
+# Logic from `testthat::on_ci()`
+if (isTRUE(as.logical(Sys.getenv("CI")))) {
+  rsconnect::setAccountInfo(
+    name   = account,
+    token  = Sys.getenv("SHINYAPPS_TOKEN"),
+    secret = Sys.getenv("SHINYAPPS_SECRET")
+  )
+}
+
+
+deploy_app <- function(
+  app_dir,
+  name,
+  ...
+) {
+  cat("\n\n\n")
+  message("Deploying: ", paste0(tail(readLines(file.path(app_dir, "app.R")), 2), collapse = "\n"))
+
+  rsconnect::deployApp(
+    appDir = app_dir,
+    appName = name,
+    server = server,
+    account = account,
+    forceUpdate = TRUE,
+    ...
+  )
+}
+
+
+
+# code_chunk(OSUICode::get_example("tabler/button"), "r")
+pkg_root <-  rprojroot::find_package_root_file("")
+app_list <- lapply(dir(pkg_root, pattern = "\\.Rmd$"), function(rmd) {
+  rmd_lines <- readLines(rmd, warn = FALSE)
+  is_example_code <- grepl("OSUICode::get_example", rmd_lines, fixed = TRUE)
+  if (!any(is_example_code)) return(NULL)
+  example_lines <- rmd_lines[is_example_code]
+  # collect app name like `"tabler/button"`
+  app_names <- sub('^.*OSUICode::get_example\\("([^"]+)".*$', "\\1", example_lines)
+  # collect package name used (if any) otherwise use `"OSUICode"`
+  package_names <- rep("OSUICode", length(app_names))
+  has_package <- grepl("package\\s*=", example_lines)
+  package_names[has_package] <- sub('^.*OSUICode::get_example\\(.*package\\s*=\\s*"([^"]+)".*$', "\\1", example_lines[has_package])
+
+  data.frame(
+    app_name = app_names,
+    package_name = package_names
+  )
+})
+
+# Find all unique combinations of apps and packages
+apps <- unique(do.call(rbind, app_list))
+message("Applications:")
+print(apps)
+
+library_deps <- desc::desc(file.path(pkg_root, "DESCRIPTION"))$get_deps()
+library_code <- paste0(
+  "# Insert code to trick `manifest.json` creation\n",
+  "if (FALSE) {",
+    "\n",
+    paste0("  library(", library_deps$package, ")", collapse = "\n"),
+    "\n",
+  "}"
+)
+
+apps$i <- seq_len(nrow(apps))
+
+# Deploy in parallel
+doParallel::registerDoParallel(cores = 3)
+runInParallel <- TRUE
+
+for (i in 1:3) {
```

Comment:
is 3 arbitrary?

Reply:
Yes?

Thought process...

There's a small chance (let's say 1%) that things go wrong just because. In that case, should try to deploy the app again (especially when working with 100 apps). By making `k = 3`, the probability of things going wrong by chance three times in a row for any of the applications is pretty slim: `p = 100 * (0.01^3) = 0.0001`. 

If it _does_ fail three times in a row, then there is probably a deploy / app issue.

Reply:
I also noticed sometimes irreproducible deployment errors. That makes sense though.